src/PrestaShopBundle/Resources/config/services/adapter/form.yml

```diff
@@ -93,3 +93,8 @@ services:
       - '@prestashop.adapter.attribute.repository.attribute_repository'
       - '@prestashop.core.product.combination.name_builder.combination_name_builder'
       - '@=service("prestashop.adapter.legacy.context").getLanguage().id'
+
+  prestashop.adapter.form.choice_provider.contact_type_choice_provider:
```

Comment:
This definition should be in the `choice_provider.yml` file, it's not a form type it's a choice provider

Reply:
i guess i can move them from adapter to core aswell, then it should be in choice_provider.yml

Reply:
Nearly all choice provider are used by form types actually, but we still separate them because we have a lot of choice providers and a lot of form types So we try to keep things (a bit) clean 😅